libs/gamify/badges/src/lib/badges.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { ApplicationsService } from '@gamify/application';
+import { BadRequestException, Body, Controller, Delete, Get, NotFoundException, Param, ParseIntPipe, Post, Put } from '@nestjs/common';
+import { BadgesService } from './badges.service';
+import { CreateBadgeInput } from './dto/create-badge.input';
+import { UpdateBadgeInput } from './dto/update-badge.input';
+
+@Controller('badges')
+export class BadgesController {
+  constructor(private badgesService: BadgesService, private applicationsService: ApplicationsService) {}
+
+  @Post()
+  async create(@Body() createBadgeInput: CreateBadgeInput) {
+    if (await this.applicationsService.findOne(createBadgeInput.applicationId) === null) {
+      throw new BadRequestException('Application not found');
```

Comment:
```suggestion
      throw new BadRequestException('Batch not found');
```

Reply:
`Badge`* presumably.
But I think the message is correct looking at the scenario at which it throws; if no application with said `id` is found.

Reply:
Cas is right in this scenario, a badge can't be created if it doesn't have a valid application id